### ISSUE TITLE
Awkward single-digit entry required in Page Setup field

### ISSUE THREAD

**Issue by scorster** [Jojo-Schmitz/MuseScore#331]
In **Format>Page Settings** (and some other places) the value fields don't accept a number entered, like: **2.25**

    • Select any field in Page Settings, so all digits are highlighted
    • Type **2.25**

**Unexpected result:** MuseScore enters **2.00** (It only captures the first digits typed)

With persistence you can enter 2.25 by dealing with digits individually: 

    • First select a target digit
    • Then type a single-digit entry. 
    • Continue doing this until you have 2.25.

Here's a related issue:

    • Starting with the value of 2.25
    • Select the **25** or **.25** 
    • Then and type 35

**Unexpected result:** 2.30; 2.35 is the expected results

Also related:

It's only possible to (click and) enter a number directly before the decimal point. All other entries are possibly only where a single digit is preselected.

Pretty strange stuff. And it's been this way at least since 3.6.2, at least circa 1999. Fixed in MuseScore 4.

MacOS Ventura 13.5.2

**Comment by Jojo-Schmitz** [comment]
Ypou can't by change pinpoint the change that fixed it for Mu4, can you?

**Comment by scorster** [comment]
> *You can't by chance pinpoint the change that fixed it for Mu4, can you?*

Hi Jojo, 

Are you asking if I know the location of the code? If so, sorry. No I don't.

Related posts from .org:
https://musescore.org/en/node/281445

https://musescore.org/en/node/290931

https://musescore.org/en/node/290952

**Comment by Jojo-Schmitz** [comment]
None of these help, unforunately. Netiher did comparing the code for page settings between 3.x and masster help

**Comment by worldwideweary** [comment]
I think this is just how Qt deals with the forcing of decimal size. If you check Mu4, you don't see "forced" decimals before typing them in the page staff spacing, and you can just type them in. In Mu3, however, the 3 digits of the "mantissa" are forced and "in the way" of typing. Notice Mu4 still has that behavior of forced zeros in the margins though...

You could just remove the digit enforcement if that's what you guys want.

**Comment by Jojo-Schmitz** [comment]
How?

**Comment by worldwideweary** [comment]
Not sure, I'll take a look.

**Comment by worldwideweary** [comment]
Looks like Qt doesn't have this by default according to for example https://stackoverflow.com/questions/62910529/override-following-digits-in-qspinbox

Either 1) Get used to going backwards, i.e. start on the right-hand side and delete + insert, or
2) Overload the thing to allow for it. That's probably what Mu4 did somewhere: implemented their own doublespinbox

**Comment by scorster** [comment]
In MuseScore 3.6.2 **Format>Style>Score>Enable indentation on the first system** we have **First system indentation** 
Its default value is 5.25 and it doesn’t require any special “edit from the rear” maneuvers

In **Format>Style>Measure Numbers** except for Interval” all of the nummeric fields are double decimal real numbers in the format x.xx ... and all are editable without any special “edit from the rear” maneuvers.

Same with  **Format>Style>System**

Makes me wonder if they simply reference a different control than the doublespinner in the Page Settings dialog.

**Comment by Jojo-Schmitz** [comment]
Yes, there's certainly some inconsistency in behavoir, but so far I haven't been able to pinpoint the reason

**Comment by Jojo-Schmitz** [comment]
See #858 for a partial fix
